homeassistant/components/zamg/sensor.py

```diff
@@ -241,36 +267,70 @@ def setup_platform(
     )
 
 
-class ZamgSensor(SensorEntity):
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the ZAMG sensor platform."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    if coordinator.config_entry is None:
+        coordinator.config_entry = entry
+
+    async_add_entities(
+        [
+            ZamgSensor(coordinator, entry.title, description)
```

Comment:
You can have `entry` as an attribute of `coordinator`. This can eliminate the extra argument. Make a separate class for `coordinator` in `coordinator.py`.

Reply:
I am not sure how to implement your suggestion.. maybe you can help out
currently it's not nice with the extra argument, but it's working ;-)